packages/handlefi/src/input-filters.ts

```diff
@@ -0,0 +1,190 @@
+import { type SwapActionParams } from '@rabbitholegg/questdk'
+import { getAddress, zeroAddress } from 'viem'
+import { buildV2PathQueryWithCase } from './utils'
+import {
+  PARASWAP_ABI,
+  V2_ROUTER_ABI,
+  HPSM2_ABI,
+  HLP_CURVE_V2_ABI,
+  HLP_BALANCER_ABI,
+  CURVE_FACTORY_ABI,
+} from './abi'
+import { WETH } from './constants'
+
+export function getParaSwapFilter(params: SwapActionParams) {
+  const { tokenIn, tokenOut, amountIn, amountOut } = params
+  const internalEthAddress = '0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE'
+  const tokenInUsed = tokenIn === zeroAddress ? internalEthAddress : tokenIn
+  const tokenOutUsed = tokenOut === zeroAddress ? internalEthAddress : tokenOut
+
+  return {
+    $abi: PARASWAP_ABI,
+    $or: [
+      {
+        // simpleswap, directUniV3Swap, directCurveSwap
+        data: {
+          fromToken: tokenInUsed,
+          fromAmount: amountIn,
+          toAmount: amountOut,
+          toToken: tokenOutUsed,
+          partner: '0xFa2c1bE677BE4BEc8851D1577B343F7060B51E3A',
```

Comment:
What do you think about making this a param from the function call? I would either do that or hardcode it to a const.